base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -473,32 +473,53 @@ impl LMDBDatabase {
     }
 
     fn set_metadata(
-        &mut self,
+        &self,
         txn: &WriteTransaction<'_>,
         k: MetadataKey,
         v: MetadataValue,
     ) -> Result<(), ChainStorageError>
     {
         lmdb_replace(txn, &self.metadata_db, &(k as u32), &v)?;
-        self.is_mem_metadata_dirty = true;
         Ok(())
     }
 
-    fn insert_orphan_block(&mut self, txn: &WriteTransaction<'_>, block: &Block) -> Result<(), ChainStorageError> {
+    fn insert_orphan_block(&self, txn: &WriteTransaction<'_>, block: &Block) -> Result<(), ChainStorageError> {
         let k = block.hash();
-        if lmdb_exists(txn, &self.orphans_db, k.as_slice())? {
-            return Ok(());
-        }
-
```

Comment:
we did we remove this check?

Reply:
Yeah it now errors if the orphan has already been inserted and checks for existence were added where necessary (i.e. add_block)